### PR TITLE
[COM-158] add commerce catalogId to queryBuilder

### PR DESCRIPTION
**Pull request by dlafreniere**
[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

**Comment by dlafreniere** [comment]
@fbeaudoincoveo the public documentation (https://bitbucket.org/coveord/doc_jekyll-public-site/src/master/collections/_javascript-search-framework/QueryBuilder.md) has to be updated as well to match this.

File: src/rest/Query.ts [hunk 1]
```diff
@@ -25,6 +25,16 @@ export interface IUserActionsRequest {
   tagViewsOfUser: string;
 }
 
+/**
+ * The information about an e-commerce search request.
```
Comment (fbeaudoincoveo) [inline src/rest/Query.ts]: Suggestion:

```
A Search API commerce request.
```

File: src/rest/Query.ts [hunk 2]
```diff
@@ -25,6 +25,16 @@ export interface IUserActionsRequest {
...
+ */
+export interface ICommerceRequest {
+  /**
+   * The id of the catalog we would like to query.
```
Comment (fbeaudoincoveo) [inline src/rest/Query.ts]: Suggestion:

```
The unique identifier of the catalog to query.

**Example:** `46bc4275-e613-4dd5-b1ea-3e5aca1bcd9d`
```

(I'm assuming catalog IDs are generated as GUIDs; if this is not the case, please replease the example I suggest by something that makes sense :) ).

File: src/rest/Query.ts [hunk 3]
```diff
@@ -273,4 +283,8 @@ export interface IQuery {
    *  - 6318b0c6-9397-4d70-b393-cf4770fd1bab
    */
   userActions?: IUserActionsRequest;
+  /**
+   * A request for a Commerce query.
```
Comment (fbeaudoincoveo) [inline src/rest/Query.ts]: Suggestion:

```
The commerce request to execute.
```

File: src/ui/Base/QueryBuilder.ts [hunk 4]
```diff
@@ -324,6 +324,10 @@ export class QueryBuilder {
    * A request to retrieve user actions in the query response.
    */
   public userActions: IUserActionsRequest;
+  /**
+   * A request for a Commerce query.
```
Comment (fbeaudoincoveo) [inline src/ui/Base/QueryBuilder.ts]: Commerce -> commerce

**Comment by fbeaudoincoveo** [comment]
> @fbeaudoincoveo the public documentation (https://bitbucket.org/coveord/doc_jekyll-public-site/src/master/collections/_javascript-search-framework/QueryBuilder.md) has to be updated as well to match this.

I'll actually delete this topic; it is incredibly outdated and quite useless: people should use the generated reference instead :)

**Pull request merged**